// storybook/dist/telemetry/index.js
import CJS_COMPAT_NODE_URL_1oy2eo17p5a from 'node:url';
import CJS_COMPAT_NODE_PATH_1oy2eo17p5a from 'node:path';
import CJS_COMPAT_NODE_MODULE_1oy2eo17p5a from "node:module";

var __filename = CJS_COMPAT_NODE_URL_1oy2eo17p5a.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_1oy2eo17p5a.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_1oy2eo17p5a.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  ErrorCollector,
  addToGlobalContext,
  cleanPaths,
  computeStorybookMetadata,
  detectAgent,
  getStorybookMetadata,
  isExampleStoryId,
  isTelemetryModuleEnabled,
  isTelemetryStateResolved,
  metaFrameworks,
  onPayloadError,
  oneWayHash,
  removeAnsiEscapeCodes,
  sanitizeAddonName,
  sanitizeError,
  setTelemetryEnabled,
  telemetry
} from "../_node-chunks/chunk-WMHXN3YR.js";
import {
  SESSION_TIMEOUT,
  getSessionId
} from "../_node-chunks/chunk-QGUN5POW.js";
import "../_node-chunks/chunk-DTA6RZFF.js";
import "../_node-chunks/chunk-WKCYHYMS.js";
import "../_node-chunks/chunk-BJVLY7MM.js";
import "../_node-chunks/chunk-IYHX5ZFZ.js";
import "../_node-chunks/chunk-FZZ2KQY7.js";
import {
  getLastEvents,
  getPrecedingUpgrade,
  isStoryCreatedByAISetup,
  isWithinInitialSession
} from "../_node-chunks/chunk-2VPOESW7.js";
import "../_node-chunks/chunk-4OKXR4H4.js";
import "../_node-chunks/chunk-C6PIYNCR.js";
import "../_node-chunks/chunk-OHGHOIHA.js";
import "../_node-chunks/chunk-EGC27Z5J.js";
import "../_node-chunks/chunk-MRJ66S5A.js";
import "../_node-chunks/chunk-MJL2XQWP.js";
import "../_node-chunks/chunk-6FSCRCDN.js";
import "../_node-chunks/chunk-JQLGW5TU.js";
import "../_node-chunks/chunk-REFVGE2Y.js";
import "../_node-chunks/chunk-QJO6IJ3P.js";
import "../_node-chunks/chunk-EBFUVDU3.js";
export {
  ErrorCollector,
  SESSION_TIMEOUT,
  addToGlobalContext,
  cleanPaths,
  computeStorybookMetadata,
  detectAgent,
  getLastEvents,
  getPrecedingUpgrade,
  getSessionId,
  getStorybookMetadata,
  isExampleStoryId,
  isStoryCreatedByAISetup,
  isTelemetryModuleEnabled,
  isTelemetryStateResolved,
  isWithinInitialSession,
  metaFrameworks,
  onPayloadError,
  oneWayHash,
  removeAnsiEscapeCodes,
  sanitizeAddonName,
  sanitizeError,
  setTelemetryEnabled,
  telemetry
};
