// storybook/dist/common/index.js
import CJS_COMPAT_NODE_URL_1oy2eo17p5a from 'node:url';
import CJS_COMPAT_NODE_PATH_1oy2eo17p5a from 'node:path';
import CJS_COMPAT_NODE_MODULE_1oy2eo17p5a from "node:module";

var __filename = CJS_COMPAT_NODE_URL_1oy2eo17p5a.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_1oy2eo17p5a.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_1oy2eo17p5a.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  DEFAULT_FILES_PATTERN,
  FileSystemCache,
  HandledError,
  JsPackageManager,
  JsPackageManagerFactory,
  PackageManagerName,
  builderPackages,
  cache,
  checkAddonOrder,
  checkRef,
  clearVitePlusCache,
  commonGlobOptions,
  compilerPackages,
  createFileSystemCache,
  createLogStream,
  doesVariableOrFunctionDeclarationExist,
  executeCommand,
  executeCommandSync,
  executeNodeCommand,
  extractFrameworkPackageName,
  extractRenderer,
  filterPresetsConfig,
  findConfigFile,
  formatFileContent,
  frameworkPackages,
  frameworkToBuilder,
  frameworkToRenderer,
  getAbsolutePathWrapperAsCallExpression,
  getAbsolutePathWrapperName,
  getAddonNames,
  getAutoRefs,
  getBuilderOptions,
  getConfigInfo,
  getDirectoryFromWorkingDir,
  getEnvConfig,
  getFieldsForGetAbsolutePathWrapper,
  getFrameworkName,
  getInterpretedFile,
  getPackageDetails,
  getPresets,
  getPrettier,
  getPrettyPackageManagerName,
  getPreviewBodyTemplate,
  getPreviewHeadTemplate,
  getRefs,
  getRendererName,
  getStoryId,
  getStoryTitle,
  getStorybookConfiguration,
  getStorybookInfo,
  getVitePlusVersions,
  globToRegexp,
  interpolate,
  isCorePackage,
  isGetAbsolutePathWrapperNecessary,
  isPreservingSymlinks,
  isSatelliteAddon,
  loadAllPresets,
  loadMainConfig,
  loadManagerOrAddonsFile,
  loadPreset,
  loadPreviewOrConfigFile,
  logConfig,
  normalizeStories,
  normalizeStoriesEntry,
  parseList,
  posix,
  readTemplate,
  removeAddon,
  rendererPackages,
  resolveAddonName,
  resolveImport,
  resolvePathInStorybookCache,
  satisfies,
  scanAndTransformFiles,
  serverRequire,
  setupAddonInConfig,
  supportedExtensions,
  syncPreviewAddonsWithMainConfig,
  syncStorybookAddons,
  temporaryDirectory,
  temporaryFile,
  transformImportFiles,
  validateConfigurationFiles,
  validateFrameworkName,
  wrapValueWithGetAbsolutePathWrapper
} from "../_node-chunks/chunk-WKCYHYMS.js";
import "../_node-chunks/chunk-BJVLY7MM.js";
import {
  MIN_SUPPORTED_NODE_DESCRIPTION,
  MIN_SUPPORTED_NODE_VERSIONS,
  formatMinVersion,
  isNodeVersionSupported,
  versions_default
} from "../_node-chunks/chunk-IYHX5ZFZ.js";
import "../_node-chunks/chunk-FZZ2KQY7.js";
import {
  findFilesUp,
  getProjectRoot,
  invalidateProjectRootCache,
  isCI,
  isWebContainer,
  loadEnvs,
  nodePathsToArray,
  normalizeStoryPath,
  optionalEnvToBoolean,
  stringifyEnvs,
  stringifyProcessEnvs
} from "../_node-chunks/chunk-4OKXR4H4.js";
import "../_node-chunks/chunk-OHGHOIHA.js";
import {
  importMetaResolve,
  importModule,
  resolvePackageDir,
  safeResolveModule
} from "../_node-chunks/chunk-EGC27Z5J.js";
import {
  groupBy,
  invariant
} from "../_node-chunks/chunk-MRJ66S5A.js";
import "../_node-chunks/chunk-MJL2XQWP.js";
import "../_node-chunks/chunk-6FSCRCDN.js";
import "../_node-chunks/chunk-JQLGW5TU.js";
import "../_node-chunks/chunk-REFVGE2Y.js";
import "../_node-chunks/chunk-QJO6IJ3P.js";
import "../_node-chunks/chunk-EBFUVDU3.js";
export {
  DEFAULT_FILES_PATTERN,
  FileSystemCache,
  HandledError,
  JsPackageManager,
  JsPackageManagerFactory,
  MIN_SUPPORTED_NODE_DESCRIPTION,
  MIN_SUPPORTED_NODE_VERSIONS,
  PackageManagerName,
  builderPackages,
  cache,
  checkAddonOrder,
  checkRef,
  clearVitePlusCache,
  commonGlobOptions,
  compilerPackages,
  createFileSystemCache,
  createLogStream,
  doesVariableOrFunctionDeclarationExist,
  executeCommand,
  executeCommandSync,
  executeNodeCommand,
  extractFrameworkPackageName,
  extractRenderer,
  filterPresetsConfig,
  findConfigFile,
  findFilesUp,
  formatFileContent,
  formatMinVersion,
  frameworkPackages,
  frameworkToBuilder,
  frameworkToRenderer,
  getAbsolutePathWrapperAsCallExpression,
  getAbsolutePathWrapperName,
  getAddonNames,
  getAutoRefs,
  getBuilderOptions,
  getConfigInfo,
  getDirectoryFromWorkingDir,
  getEnvConfig,
  getFieldsForGetAbsolutePathWrapper,
  getFrameworkName,
  getInterpretedFile,
  getPackageDetails,
  getPresets,
  getPrettier,
  getPrettyPackageManagerName,
  getPreviewBodyTemplate,
  getPreviewHeadTemplate,
  getProjectRoot,
  getRefs,
  getRendererName,
  getStoryId,
  getStoryTitle,
  getStorybookConfiguration,
  getStorybookInfo,
  getVitePlusVersions,
  globToRegexp,
  groupBy,
  importMetaResolve,
  importModule,
  interpolate,
  invalidateProjectRootCache,
  invariant,
  isCI,
  isCorePackage,
  isGetAbsolutePathWrapperNecessary,
  isNodeVersionSupported,
  isPreservingSymlinks,
  isSatelliteAddon,
  isWebContainer,
  loadAllPresets,
  loadEnvs,
  loadMainConfig,
  loadManagerOrAddonsFile,
  loadPreset,
  loadPreviewOrConfigFile,
  logConfig,
  nodePathsToArray,
  normalizeStories,
  normalizeStoriesEntry,
  normalizeStoryPath,
  optionalEnvToBoolean,
  parseList,
  posix,
  readTemplate,
  removeAddon,
  rendererPackages,
  resolveAddonName,
  resolveImport,
  resolvePackageDir,
  resolvePathInStorybookCache,
  safeResolveModule,
  satisfies,
  scanAndTransformFiles,
  serverRequire,
  setupAddonInConfig,
  stringifyEnvs,
  stringifyProcessEnvs,
  supportedExtensions,
  syncPreviewAddonsWithMainConfig,
  syncStorybookAddons,
  temporaryDirectory,
  temporaryFile,
  transformImportFiles,
  validateConfigurationFiles,
  validateFrameworkName,
  versions_default as versions,
  wrapValueWithGetAbsolutePathWrapper
};
